### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by HanningtoneM**
## objectives :green_circle: 

- Use preprocessed HTML file with embedded Ruby code.
- Use layouts and templates for shared content.

## completed :heavy_check_mark: 

- Implement the design from the sneak peek wireframes.
- Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1 +1,11 @@
...
\ No newline at end of file
+<h1 class="post_heading">This is all posts of users</h1>
+
+<div id="postTemplate">
+   <%= render partial: "shared/single_user", locals: { user: @user } %>
+<% @posts.each do |post| %>
+     <%= render partial: "shared/single_post", locals: { post: post } %>
+     <%= render partial: "shared/collection_comments", locals: { post: post, user: @user } %>
+<% end %>
+<%= button_to "Paginations"%>
```
Comment (Davidosky007) [inline app/views/posts/index.html.erb]: * Kindly use `post.recent_comments`  method to render 5 most recent comments, you have this methods created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md). to render the user's 3 most recent posts. [as mentioned in the project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#:~:text=Use%20methods%20that%20you%20have%20created%20in%20the%20Blog%20App%20%2D%20processing%20data%20in%20models.)

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1 +1,7 @@
-<h1>This is for all users</h1>
\ No newline at end of file
+<h1 class="post_heading">This is for all users</h1>
+
+<div id="users">
+  <% @users.each do |user| %>
+    <%= render partial: "shared/single_user", locals: { user: user } %>
+  <% end %>
+</div>
```
Comment (Davidosky007) [inline app/views/users/index.html.erb]: * Kindly use `#recent_post` method to render the user's 3 most recent posts of a user, you have this methods created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md). to render the user's 3 most recent posts. [as mentioned in the project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#:~:text=Use%20methods%20that%20you%20have%20created%20in%20the%20Blog%20App%20%2D%20processing%20data%20in%20models.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @HanningtoneM 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Well written PR description 👍 
* Good looking readme file 👍 
* Proper use of Gitflow 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @HanningtoneM, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Requested changes ✔
- Linters ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**